js-jasmine/src/gilded_rose.js

```diff
@@ -10,57 +10,120 @@ class Shop {
   constructor(items=[]){
     this.items = items;
   }
-  updateQuality() {
-    for (var i = 0; i < this.items.length; i++) {
-      if (this.items[i].name != 'Aged Brie' && this.items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
-        if (this.items[i].quality > 0) {
-          if (this.items[i].name != 'Sulfuras, Hand of Ragnaros') {
-            this.items[i].quality = this.items[i].quality - 1;
-          }
-        }
-      } else {
-        if (this.items[i].quality < 50) {
-          this.items[i].quality = this.items[i].quality + 1;
-          if (this.items[i].name == 'Backstage passes to a TAFKAL80ETC concert') {
-            if (this.items[i].sellIn < 11) {
-              if (this.items[i].quality < 50) {
-                this.items[i].quality = this.items[i].quality + 1;
-              }
-            }
-            if (this.items[i].sellIn < 6) {
-              if (this.items[i].quality < 50) {
-                this.items[i].quality = this.items[i].quality + 1;
-              }
-            }
-          }
-        }
+
+  reduceQuality(input) {
+    if (0 < input.quality && input.quality < 51){
+    return input.quality --
+    }
+  }
+
+  increaseQuality(input) {
+    if (0 <= input.quality && input.quality < 50){
+    return input.quality ++
+    }
+  }
+        
+  reduceSellIn(input){
+    if (this.isItemSulfuras(input) === false && input.sellIn > 0){
+      return input.sellIn --
+    }
+  }
+
+  isItemSulfuras(input){
+    if('Sulfuras, Hand of Ragnaros' === input.name){
+      return true
+    }
+    else
+    return false
```

Comment:
- I think you can try shorten the item validator methods further by returning input.name === true